src/main/java/tictactoe/Minimax.java

```diff
@@ -0,0 +1,81 @@
+package tictactoe;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class Minimax {
+
+    public int chooseMove
+            (Board board, String currentPlayersMark, String opponentsMark) {
+
+        var copyBoard = board.copyBoard();
+        return findBestMove(copyBoard, 0, currentPlayersMark, opponentsMark);
+    }
+
+    private int findBestMove
+            (Board board, int depth, String currentPlayersMark, String opponentsMark) {
+        var movesAndScores = new HashMap<Integer, Integer>();
+        var availableSquares = board.availableSquares(currentPlayersMark, opponentsMark);
+
+        if (board.isComplete(currentPlayersMark, opponentsMark)) {
```

Comment:
Not really relevant to the code here - but feels odd that you need to pass the player's marks to find out if a board is complete.

Reply:
The isComplete method checks whether the board isFull (a tie) or if a player has won, by calling the winningLineExists method. This method uses the player's marks to see if any player has a a winning line.